crates/starknet-os-types/src/contract_class.rs

```diff
@@ -0,0 +1,218 @@
+use std::cell::OnceCell;
+use std::rc::Rc;
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::hash::GenericClassHash;
+
+pub type CairoLangCasmClass = cairo_lang_starknet_classes::casm_contract_class::CasmContractClass;
+pub type BlockifierCasmClass = blockifier::execution::contract_class::ContractClassV1;
+
+#[derive(thiserror::Error, Debug)]
+pub enum ContractClassError {
+    #[error("Internal error: no type conversion is possible to generate the desired effect.")]
+    NoPossibleConversion,
+
+    #[error("Could not build Blockifier contract class")]
+    BlockifierConversionError,
+
+    #[error(transparent)]
+    SerdeError(#[from] serde_json::Error),
+}
+
+/// A generic contract class that supports conversion to/from the most commonly used
+/// contract class types in Starknet and provides utility methods.
+/// Operations are implemented as lazily as possible, i.e. we only convert
+/// between different types if strictly necessary.
+/// Fields are boxed in an RC for cheap cloning.
+#[derive(Debug, Clone)]
+pub struct GenericCasmContractClass {
+    blockifier_contract_class: OnceCell<Rc<BlockifierCasmClass>>,
+    cairo_lang_contract_class: OnceCell<Rc<CairoLangCasmClass>>,
+    serialized_class: OnceCell<Vec<u8>>,
+    class_hash: OnceCell<GenericClassHash>,
```

Comment:
This type is small, is it worth the overhead of `OnceCell`?

Reply:
The memory footprint is small, the computation to get the value is not.